app/api/events.py

```diff
@@ -556,12 +556,14 @@ def before_update_object(self, event, data, view_kwargs):
         :param view_kwargs:
         :return:
         """
-        # 1 - call validate_date if either event start_at or end_at is updated
-        # 2 - call validate_date if drafted event is being published
-        # 3 - call validate_date if event is being restored
-        if (data.get('starts_at') != event.starts_at or data.get('ends_at') != event.ends_at) or \
-                (event.state == "draft" and data.get('state') == "published") or \
-                (event.deleted_at and not data.get('deleted_at')):
+        # is either event start_at or end_at is updated
+        is_date_updated = (data.get('starts_at') != event.starts_at or
+                           data.get('ends_at') != event.ends_at)
+        # is drafted event is being published
+        is_draft_published = (event.state == "draft" and data.get('state') == "published")
+        # is deleted event is being restored
```

Comment:
Remove comments now, variable names are seld explanatory

Reply:
Made the changes. Please have a look